lib/nats/io/client.rb

```diff
@@ -61,6 +61,9 @@ def connect(uri=nil, opts={})
   # the connection is manually closed then it will reach the CLOSED
   # connection state after which it will not reconnect again.
   module Status
+    # When client was just initialized.
+    NOT_YET_CONNECTED = -1
```

Comment:
rather than adding a new state here I would rather use a different heuristic to check whether having to connect, I something like this maybe: https://github.com/nats-io/nats-pure.rb/commit/530ad2f0be001741fda97a9b938f5ce40898cfac